src/cargo/ops/cargo_fetch.rs

```diff
@@ -76,6 +76,7 @@ pub fn fetch<'a>(
     }
 
     packages.get_many(to_download)?;
+    crate::core::gc::auto_gc(config);
```

Comment:
When compiling, we gc first.  When fetching, we gc after.  Which should we do generally and why the deviation?

Reply:
I don't think there is really a before/after difference here. Gc always happens after downloading. The sequence is generally:

* `cargo fetch` — fetch index, resolve, download, gc
* `cargo build` — fetch index, resolve, download, gc, start compiling

It is not done after compiling to make it easier to deal with error handling (I felt like it was better to not co-mingle the two, or to require compiling to succeed, or to perform some action after compiling failed).

Reply:
I think my comment was future-looking to when we do `target/` GC but we can look into it further then